### ISSUE TITLE
Problem with pom.xml

### ISSUE THREAD

**Issue by CrueDE** [coylz/TobiiStreamEngineForJava#13]
I hope you guys can help me. When I use the project in Netbeans it works and prints out the gaze position in a extra window when I run the TobiiDemo class. But when I click on clean and build, the jar in the target folder does nothing. The class by itself in another project without the gaze function works fine. Any idea how to fix that?

**Comment by Tohaker** [comment]
It appears as though the compiled JAR doesn't have a manifest, as I can't run it from the command line either. However, the demo is not really intended to be run on its own. Does the imported Tobii class work fine for you in another project?

**Comment by CrueDE** [comment]
When I try to use Tobii.java in another project it will say that it does not find the needed library.

**Comment by CrueDE** [comment]
TobiiDemo.java without the Gaze Method also works fine in another project packed into a jar.

**Comment by Tohaker** [comment]
Could you post the exact error it shows? It sounds like it just can't find the driver for the eye tracker

**Comment by CrueDE** [comment]
![image](https://user-images.githubusercontent.com/84240244/127034318-5fbff33a-e094-46a2-a9e3-a89496f0f72c.png)

It cant find the dll, because it is not in the folder of that extra project.

Running the other project downloaded from this github with the dll inside, it runs fine in Netbeans but won't run when I click on the jar in the target folder after I clicked on "clean and build" in Netbeans.

**Comment by CrueDE** [comment]
![image](https://user-images.githubusercontent.com/84240244/127035050-b94fa00c-d416-4f1b-8f05-d41592660ed2.png)

This is the output from the clean and build process, no error.

**Comment by Tohaker** [comment]
Do you have the dll in that location? If it can't be found for your other project, it's likely a system setup issue, which may be on us to document better.

**Comment by CrueDE** [comment]
To summarize it all:

- I downloaded the project from this github, modified the TobiiDemo.java to print out the gaze position in a jframe

- The modified TobiiDemo.java class with the Tobii.gazePosition() medthod removed works fine in another project, after beeing packed into a jar

- The original project from github with the modified TobiiDemo.java class works as expected with no errors inside of Netbeans, but when I build it into the jar inside the target folder, it wont run. No error, just nothing. Java also does not show up in task manager as running.

- I hope I understood you right that you then wanted me to put the TobiiDemo Class and the Tobii Class inside a new project and to test it then. That produced the error with the dll. I checked the folder mentioned in that error in the image I posted above and there is a dll.

![image](https://user-images.githubusercontent.com/84240244/127202872-05ac974a-3f77-4547-a29f-9e60288924ea.png)

You said something about a manifest before, could it be that that maven compiler plugin doesn't know what the main class is? In my case that would be the TobiiDemo Class

**Comment by CrueDE** [comment]
Hello everyone, if anyone has an idea how to fix this, please help! I really love this project and use it all the time!

**Comment by schwabdidier** [comment]
Hello @CrueDE, sorry the best able to help you is @kuhlkrein certainly but he is in vacation. I try to help as I worked on the first version (but it was certainly 3 years ago).

Can you specify on which OS you are ? I assume it is Windows ?  The problem seems to come from jar generation. To be short, Java code which is in a jar is not exactly the same as java code in netbeans or any other IDE (ie .class in a folder). for instance, you can directly put path and you have to go through methods like getResourceAsStream . Maybe this link can help you ? https://stackoverflow.com/questions/1611357/how-to-make-a-jar-file-that-includes-dll-files Hope it helps, please, give me a feedback.

**Comment by CrueDE** [comment]
Hello @schwabdidier and @Tohaker, great news! I got it running!
I added this to the pom.xml

![image](https://user-images.githubusercontent.com/84240244/127925101-d893ae8b-2f69-4fef-86dc-69af4a653bc3.png)

Now when clicking on the jar the JFrame is created, the libraries are being loaded and the eyes are being tracked correctly!
Source for the code: https://maven.apache.org/plugins/maven-shade-plugin/examples/executable-jar.html
